### PR TITLE
DOC: clarify "location" in docstring of cKDTree.query

### PR DESCRIPTION
**Pull request by PinkShnack**
#### Reference issue
Perhaps a small thing, but I think for the cKDTree.query return, the `i` should be unambiguously described as an index. 
The current description: "The locations of the neighbors..." implies to me that it returns the coordinate locations of the neighbours, and I spent a few minutes wondering why my 2D coordinates were not being returned until I realised it was the index.

#### What does this implement/fix?
It may clarify at a glance what this method is returning.

### CODE REVIEW

File: scipy/spatial/ckdtree.pyx [hunk 1]
```diff
@@ -706,7 +706,7 @@ cdef class cKDTree:
             When k == 1, the last dimension of the output is squeezed.
             Missing neighbors are indicated with infinite distances.
         i : ndarray of ints
-            The locations of the neighbors in ``self.data``.
+            The index locations of the neighbors in ``self.data``.
```
Comment (peterbell10) [inline scipy/spatial/ckdtree.pyx:709]: I think you can drop the word "locations". Also, can you update the [KDTree.query](https://github.com/scipy/scipy/blob/2a1dc2ae8d2c944209f3053d1285dbe5edaff6cc/scipy/spatial/kdtree.py#L388) documentation to match.

Comment (PinkShnack) [reply scipy/spatial/ckdtree.pyx]: Done, I assumed "index" is better than "indexes" or "indices".
Also updated latest upstream commits. This PR should be good for merge.

Comment (peterbell10) [reply scipy/spatial/ckdtree.pyx]: >I assumed "index" is better than "indexes" or "indices".

Grammatically, I think it should really be "The index of each neighbor" or "The indices of the neighbours".

**Comment by peterbell10** [comment]
Thanks @PinkShnack. Merging.

**Pull request merged**